src/Detector/DetectorLibrary/DetectorLibrary.csproj

```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
```

Comment:
Change this to `netstandard2.0`

Reply:
Can I know what's the reason we want to change target framework? So this solution used some extensions from the common project and I added into it. If we changed to the netstandard2.0, it will not be compatible with common project.

Reply:
I forwarded you an email that had sent our team earlier about this. It has details about this one.

Reply:
I'm considering to keep a common project for common files within build script generator and detector, please let me know if I was wrong. For example, when we make changes within generator `public PlatformDetectorResult Detect(RepositoryContext context)`, it needs to pass in the `RepositoryContext` which both sides are using, ideally we want to call the detector api and pass in the context but the the namespace needs to be converted, one side is `BuildScriptGenerator.RepositoryContext` and the other is `Detector.RepositoryContext`. Imo the best way to keep a common files and they're both using the same RepositoyContext.

Reply:
Also files within Common.Extensions(such as FileExtension.cs, StringExtension.cs) are also duplicated in detector proj because we did't share the project. This kind of blocks me because we are using different target framework.

Reply:
So are you suggesting you want to move files to Common project so that the can be used in both Detector and BuildScriptGenerator project? Sure, sounds good.

Reply:
Yeah I'm thinking to add Common proj as dependency, and Common proj target framework is netcoreapp2.1 which are not compatible with netstandard2.0.. so we still need to move detector proj to netcoreapp2.1.